src/jit/gentree.cpp

```diff
@@ -15215,19 +15215,20 @@ bool GenTree::IsFieldAddr(Compiler* comp, GenTreePtr* pObj, GenTreePtr* pStatic,
             baseAddr  = gtOp.gtOp1;
         }
     }
+    // Check if "this" has a zero-offset annotation.
+    else if (comp->GetZeroOffsetFieldMap()->Lookup(this, &newFldSeq))
+    {
+        baseAddr = this;
+        mustBeStatic = true;
+    }
+    else if (OperGet() == GT_CNS_INT && gtIntCon.gtFieldSeq != nullptr)
+    {
+        newFldSeq = gtIntCon.gtFieldSeq;
```

Comment:
If this is a static field and `isFieldStatic(newFldSeq->m_fieldHnd))` returns true then you
 will need to set baseAddr to 'this' as well.